lib/iris/experimental/regrid.py

```diff
@@ -608,6 +631,20 @@ def _regrid_area_weighted_array(
     if not src_masked and not new_data.mask.any():
         new_data = new_data.data
 
+    # Restore axis to original order
+    if x_dim_orig is None and y_dim_orig is None:
+        pass
+    elif x_dim_orig is not None and y_dim_orig is None:
+        new_data = np.moveaxis(new_data, -1, x_dim_orig)
+    elif y_dim_orig is not None and x_dim_orig is None:
+        new_data = np.moveaxis(new_data, -1, y_dim_orig)
+    elif x_dim_orig < y_dim_orig:
+        new_data = np.moveaxis(new_data, -1, x_dim_orig)
+        new_data = np.moveaxis(new_data, -1, y_dim_orig)
+    else:
+        new_data = np.moveaxis(new_data, -2, y_dim_orig)
+        new_data = np.moveaxis(new_data, -1, x_dim_orig)
+
```

Comment:
These last two loops can be consolidated, since `new_data` will end in `[..., y, x]` in either case.

Reply:
Ok.  We'll revisit in PR  #3596 after everything has been rejigged anyway